host/tests/deepsleepClasses/power.py

```diff
@@ -0,0 +1,276 @@
+        Initializes the Power Manager class function.
+        """
+
+        self.moduleName = "power"
+        self.testConfig =  os.path.join(dir_path, "/power_testConfig.yml")
+        self.testConfig.test.execute = os.path.join(targetWorkspace, self.testConfig.test.execute)
+        self.testSuite = "L3 power"
+        
+        self.deviceProfile = ConfigRead( moduleConfigProfileFile, self.moduleName)
+        self.utMenu        = UTSuiteNavigatorClass(self.testConfig, None, session)
+        self.testSession = session
+
+        for artifact in self.testConfig.test.artifacts:
+            filesPath = os.path.join(dir_path, artifact)
+            self.utils.rsync(self.testSession, filesPath, targetWorkspace)
+
+        self.utMenu.start()
+
+    class WakeupSources(Enum):
+        WAKE_ON_VOICE = 0x00
+        WAKE_ON_PRESENCE_DETECTION = 0x01
+        WAKE_ON_BLUETOOTH = 0x02
+        WAKE_ON_WIFI = 0x03
+        WAKE_ON_IR = 0x04
+        WAKE_ON_POWER_KEY = 0x05
+        WAKE_ON_TIMER = 0x06
+        WAKE_ON_CEC = 0x07
+        WAKE_ON_LAN = 0x08
+
+    wakeup_sources_map = {
+        WakeupSources.WAKE_ON_VOICE: "Wake on Voice",
+        WakeupSources.WAKE_ON_PRESENCE_DETECTION: "Wake on Presence Detection",
+        WakeupSources.WAKE_ON_BLUETOOTH: "Wake on Bluetooth",
+        WakeupSources.WAKE_ON_WIFI: "Wake on Wifi",
+        WakeupSources.WAKE_ON_IR: "Wake on IR",
+        WakeupSources.WAKE_ON_POWER_KEY: "Wake on Power key",
+        WakeupSources.WAKE_ON_TIMER: "Wake on Timer",
+        WakeupSources.WAKE_ON_CEC: "Wake on CEC",
+        WakeupSources.WAKE_ON_LAN: "Wake on LAN"
+    }
+
+    def searchPattern(self, haystack, pattern):
+        match = re.search(pattern, haystack)
+        if match:
+            return match.group(1)
+        return None
+
+    def initialise(self):
+        """
+        Initializes the Power manager module.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        result = self.utMenu.select( self.testSuite, "Initialize Power Manger")
+
+    def getProfile(self):
+        return self.deviceProfile
+
+    def setPowerMode(self, mode:int):
+        """
+        Set Power Mode.
+
+        Args:
+            mode: int: This enumeration defines the power state.
+                        Refer to the PWRMGR_PowerState_t enum for valid options
+
+        Returns:
+            None
+        """
+
+        promptWithAnswers = [
+            {
+                "query_type": "direct",
+                "query": "Power State",
+                "input": str(mode)
+            }
+        ]
+
+        # Convert input arguments to strings and update the prompts
+        result = self.utMenu.select( self.testSuite, "Set Power State", promptWithAnswers)
+
+    def getPowerMode(self):
+        """
+        Get Power Mode.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        result = self.utMenu.select( self.testSuite, "Get Power State")
```

Comment:
this is a get function, but its not returning the mode

Reply:
Updated with a return